libs/wingsdk/src/std/resource.ts

```diff
@@ -143,11 +169,11 @@ export abstract class Resource extends Construct implements IResource {
           return;
         }
 
-        // if the object is a resource (i.e. has a "lift" method"), register a lifting between it and the host.
-        if (isResource(obj)) {
+        // if the object is a liftable resource (i.e. also has a "_addOnLift" method"), register a lifting between it and the host.
+        if (isLiftable(obj) && "_addOnLift" in obj) {
           // Explicitly register the resource's `$inflight_init` op, which is a special op that can be used to makes sure
           // the host can instantiate a client for this resource.
-          obj._addOnLift(host, [...ops, "$inflight_init"]);
+          (obj as any)._addOnLift(host, [...ops, "$inflight_init"]);
```

Comment:
why do we need `as any` here?

Reply:
TS doesn't know that _addOnLift is callable here, only that it is a property of obj. I could add a check to see that `_addOnLift` but it seemed unnecessary since it shouldn't happen. I can try changing this block to just do `if (obj instanceof Resource)` instead.